### PR TITLE
Add JAVA_OBJECT type to query result comparator

### PR DESCRIPTION
**Pull request by ebyhr**
To support UUID type in https://github.com/prestosql/presto/pull/5231

### CODE REVIEW

File: tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java [hunk 1]
```diff
@@ -261,6 +263,14 @@ private boolean timestampWithTimezoneEqual(Object actual, Object expected)
         return false;
     }
 
+    private boolean javaObjectEqual(Object actual, Object expected)
+    {
+        if (actual.getClass().equals(expected.getClass())) {
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java]: Why? Wouldn't `actual.equals(expected)` be enough?

File: tempto-core/src/test/groovy/io/prestosql/tempto/assertions/QueryResultValueComparatorTest.groovy [hunk 2]
```diff
@@ -129,6 +130,11 @@ class QueryResultValueComparatorTest
         TIMESTAMP               | Timestamp.valueOf("2015-02-16 10:10:10") | Timestamp.valueOf("2015-02-15 10:10:10") | false
         TIMESTAMP               | Timestamp.valueOf("2015-02-15 10:10:10") | Timestamp.valueOf("2015-02-16 10:10:10") | false
         TIMESTAMP               | Timestamp.valueOf("2015-02-15 10:10:10") | "a"                                      | false
+
+        JAVA_OBJECT             | UUID.fromString("50554d6e-29bb-11e5-b345-feff819cdc9f") | UUID.fromString("50554d6e-29bb-11e5-b345-feff819cdc9f") | true
+        JAVA_OBJECT             | UUID.fromString("12151fd2-7586-11e9-8f9e-2a86e4085a59") | UUID.fromString("12151fd2-7586-11e9-8f9e-2a86e4085a59") | true
+        JAVA_OBJECT             | UUID.fromString("50554d6e-29bb-11e5-b345-feff819cdc9f") | UUID.fromString("12151fd2-7586-11e9-8f9e-2a86e4085a59") | false
+        JAVA_OBJECT             | UUID.fromString("50554d6e-29bb-11e5-b345-feff819cdc9f") | "a"                       | false
```
Comment (findepi) [inline tempto-core/src/test/groovy/io/prestosql/tempto/assertions/QueryResultValueComparatorTest.groovy:139]: And a false case where types do not match (eg UUID vs String)

File: tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java (hunk 1 above)
Comment (ebyhr) [reply tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java]: Exactly, `actual.equals(expected)` is enough.

**Comment by ebyhr** [comment]
@findepi Applied comments.

File: tempto-core/src/test/groovy/io/prestosql/tempto/assertions/QueryResultValueComparatorTest.groovy [hunk 3]
```diff
@@ -129,6 +130,12 @@ class QueryResultValueComparatorTest
         TIMESTAMP               | Timestamp.valueOf("2015-02-16 10:10:10") | Timestamp.valueOf("2015-02-15 10:10:10") | false
         TIMESTAMP               | Timestamp.valueOf("2015-02-15 10:10:10") | Timestamp.valueOf("2015-02-16 10:10:10") | false
         TIMESTAMP               | Timestamp.valueOf("2015-02-15 10:10:10") | "a"                                      | false
+
+        JAVA_OBJECT             | UUID.fromString("50554d6e-29bb-11e5-b345-feff819cdc9f") | UUID.fromString("50554d6e-29bb-11e5-b345-feff819cdc9f") | true
```
Comment (findepi) [inline tempto-core/src/test/groovy/io/prestosql/tempto/assertions/QueryResultValueComparatorTest.groovy:135]: Add also a test case with `null` values (=> true)

File: tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java [hunk 4]
```diff
@@ -261,6 +263,11 @@ private boolean timestampWithTimezoneEqual(Object actual, Object expected)
         return false;
     }
 
+    private boolean javaObjectEqual(Object actual, Object expected)
+    {
+        return actual.equals(expected);
```
Comment (findepi) [inline tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java:268]: `java.util.Objects#equals` may be needed

**Review by findepi — APPROVED** [review]
one more i missed previously

File: tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java (hunk 4 above)
Comment (findepi) [reply tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java:268]: turns out not -- https://github.com/prestosql/tempto/blob/100163fb1d99dfde8c4b98fe1753e20d3ce9df56/tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java#L62-L67

Comment (ebyhr) [reply tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java:268]: @findepi Reverted to `actual.equals(expected);` from `java.util.Objects#equals`.

**Review by findepi — APPROVED** [review]

**Comment by kokosing** [comment]
Can you please rebase?

**Comment by ebyhr** [comment]
Superseded by #87 (Sorry, I was missing this PR)
